### PR TITLE
Fix the arbitrary handling of errors.

### PR DESCRIPTION
**Pull request by klopyrev**
This change adapts exceptions to HTTP errors, as appropriate. No, we don't need to catch all exceptions in the REST APIs, and return appropriate HTTP error codes. This is done automatically.

Also, the returned error message doesn't need to be JSON. It can just be the error string.

@percyliang

### CODE REVIEW

File: codalab/common.py [hunk 1]
```diff
@@ -49,6 +57,36 @@ class PermissionError(UsageError):
...
+# Listed in order of most specific to least specific.
+http_codes_and_exceptions = [
+    (httplib.FORBIDDEN, PermissionError),
+    (httplib.UNAUTHORIZED, AuthorizationError),
+    (httplib.NOT_FOUND, NotFoundError),
+    (httplib.BAD_REQUEST, UsageError),
+]
+
+
+def ExceptionToHttpError(e):
```
Comment (percyliang) [inline codalab/common.py]: Style: do exception_to_http_error

File: codalab/common.py [hunk 2]
```diff
@@ -49,6 +57,36 @@ class PermissionError(UsageError):
...
+    """
+    Returns the appropriate HTTP error code and message for the given exception.
+    """
+    for known_code, exception_type in http_codes_and_exceptions:
+        if isinstance(e, exception_type):
+            return known_code, e.message
+    return httplib.INTERNAL_SERVER_ERROR, e.message
+
+
+def HttpErrorToException(code, message):
```
Comment (percyliang) [inline codalab/common.py]: same comment as above

**Comment by percyliang** [comment]
Can you make sure that some places which depend on catching UsageError's now work with the new exception class?

Otherwise, LGTM

File: codalab/common.py (hunk 1 above)
Comment (klopyrev) [inline codalab/common.py]: Done, thanks for catching that.

File: codalab/common.py (hunk 2 above)
Comment (klopyrev) [inline codalab/common.py]: Done.

**Comment by klopyrev** [comment]
NotFoundError is a subclass of UsageError, so there shouldn't be any issues.

**Pull request merged**